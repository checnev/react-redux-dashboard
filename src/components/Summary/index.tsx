import {
  Typography,
  Link,
} from '@material-ui/core';
import { Link as LinkRoute } from 'react-router-dom';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { Skeleton } from '@material-ui/lab';
import DefinitionList from 'components/common/DefinitionList';
import type { CarReport } from 'reducers/vehicles/reports';

interface SummaryProps {
  data: CarReport | null;
}

const useStyles = makeStyles((theme: Theme) => createStyles({
  root: {
    display: 'flex',
    flexDirection: 'column',
    'height': '100%',
  },
  title: {
    marginBottom: theme.spacing(1),
  },
  definitionList: {
    flex: 1,
  },
  bottomLink: {
  },
}));

const Summary: React.FC<SummaryProps> = ({ data }: SummaryProps) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Typography
        color="primary"
        variant="h6"
        component="h2"
        className={classes.title}
      >
        Сводка
      </Typography>
      <div className={classes.definitionList}>
        <DefinitionList 
          data={[
            {
              name: 'Доход',
              value: data ? data.income : <Skeleton variant="text" />,
            },
            {
              name: 'Кол-во Аренд',
              value: data ? data.countOfLeases : <Skeleton variant="text" />,
            },
            {
              name: 'Происшествий',
              value: data ? data.incidents : <Skeleton variant="text" />,
            }
          ]}
        />
      </div>
      <div className={classes.bottomLink}>
        <Link component={LinkRoute} to="/reviews" color="primary">Отзывы</Link>
      </div>
    </div>
  );
};

export default Summary;
