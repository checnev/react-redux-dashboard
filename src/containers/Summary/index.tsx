import { useSelector } from 'react-redux';
import Summary from 'components/Summary';
import { summarySelector } from 'selectors/stats';

const SummaryContainer = () => {
  const data = useSelector(summarySelector);

  return <Summary data={data} />
};

export default SummaryContainer;
