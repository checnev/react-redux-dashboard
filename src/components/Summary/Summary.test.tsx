import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import { BrowserRouter as Router } from 'react-router-dom';
import Summary from 'components/Summary';

const summaryData = {
  income: 2000,
  countOfLeases: 20,
  incidents: 3,
};

let container: any = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('Summary render', () => {
  act(() => {
    render(<Router><Summary data={summaryData} /></Router>, container);
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
