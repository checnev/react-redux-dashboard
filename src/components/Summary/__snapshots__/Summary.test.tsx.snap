// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary render 1`] = `
"<div class=\\"makeStyles-root-1\\">
  <h2 class=\\"MuiTypography-root makeStyles-title-2 MuiTypography-h6 MuiTypography-colorPrimary\\">Сводка</h2>
  <div class=\\"makeStyles-definitionList-3\\">
    <dl class=\\"MuiGrid-root makeStyles-root-5 MuiGrid-container MuiGrid-spacing-xs-2\\">
      <div class=\\"MuiGrid-root makeStyles-ceil-6 MuiGrid-item MuiGrid-grid-xs-6\\">
        <dt class=\\"MuiTypography-root MuiTypography-body1\\">Доход:</dt>
      </div>
      <div class=\\"MuiGrid-root makeStyles-ceil-6 MuiGrid-item MuiGrid-grid-xs-6\\">
        <dd class=\\"MuiTypography-root MuiTypography-body1\\">2000</dd>
      </div>
      <div class=\\"MuiGrid-root makeStyles-ceil-6 MuiGrid-item MuiGrid-grid-xs-6\\">
        <dt class=\\"MuiTypography-root MuiTypography-body1\\">Кол-во Аренд:</dt>
      </div>
      <div class=\\"MuiGrid-root makeStyles-ceil-6 MuiGrid-item MuiGrid-grid-xs-6\\">
        <dd class=\\"MuiTypography-root MuiTypography-body1\\">20</dd>
      </div>
      <div class=\\"MuiGrid-root makeStyles-ceil-6 MuiGrid-item MuiGrid-grid-xs-6\\">
        <dt class=\\"MuiTypography-root MuiTypography-body1\\">Происшествий:</dt>
      </div>
      <div class=\\"MuiGrid-root makeStyles-ceil-6 MuiGrid-item MuiGrid-grid-xs-6\\">
        <dd class=\\"MuiTypography-root MuiTypography-body1\\">3</dd>
      </div>
    </dl>
  </div>
  <div class=\\"makeStyles-bottomLink-4\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"/reviews\\">Отзывы</a></div>
</div>"
`;
